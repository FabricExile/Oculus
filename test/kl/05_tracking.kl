require Oculus;

operator entry() {
  ovrDevice device();

  device.configureTracking(ovrTrackingCap_Orientation |
    ovrTrackingCap_MagYawCorrection |
    ovrTrackingCap_Position, 0);

  ovrTrackingState ts = device.getTrackingState(device.getTimeInSeconds());
  while(!(ts.StatusFlags & (ovrStatus_OrientationTracked | ovrStatus_PositionTracked))) {
    ts = device.getTrackingState(device.getTimeInSeconds());
  }

  report('received first tracking state.');
}