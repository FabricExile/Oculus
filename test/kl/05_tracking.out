[FABRIC:MT] Oculus: Oculus Rift initialized.
[FABRIC:MT] Oculus: Oculus Rift configured for tracking.
[FABRIC:MT] received first tracking state.
[FABRIC:MT] Oculus: Oculus Rift finalized.
